File: microservice-transaction-sample/client/src/main/java/example/client/command/GetCustomerInfoCommand.java
package example.client.command;

import example.rpc.CustomerServiceGrpc;
import example.rpc.GetCustomerInfoRequest;
import example.rpc.GetCustomerInfoResponse;
import io.grpc.ManagedChannel;
import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;

@Command(name = "GetCustomerInfo", description = "Get customer information")
public class GetCustomerInfoCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Override
  public Integer call() {
    ManagedChannel channel = Utils.getCustomerServiceChannel();
    try {
      CustomerServiceGrpc.CustomerServiceBlockingStub stub =
          CustomerServiceGrpc.newBlockingStub(channel);
      GetCustomerInfoResponse response =
          stub.getCustomerInfo(
              GetCustomerInfoRequest.newBuilder().setCustomerId(customerId).build());
      Utils.printJsonString(response);
      return 0;
    } catch (Exception e) {
      e.printStackTrace();
      return 1;
    } finally {
      Utils.shutdownChannel(channel);
    }
  }
}

File: dotnet/microservice-transactions-sample-with-shared-cluster-with-linq/Client/Commands/GetCustomerInfoCommand.cs
using System.CommandLine;
using MicroserviceTransactionsSample.Client.Commands.Binders;
using MicroserviceTransactionsSample.Rpc;
using static MicroserviceTransactionsSample.Rpc.CustomerService;

namespace MicroserviceTransactionsSample.Client.Commands;

public static class GetCustomerInfoCommand
{
    private const string Name = "GetCustomerInfo";
    private const string Description = "Get customer information";

    private const string ArgName = "id";
    private const string ArgDescription = "customer ID";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName, ArgDescription);
        var getCustomerInfoCommand = new Command(Name, Description)
        {
            customerIdArg
        };

        getCustomerInfoCommand.SetHandler(getCustomerInfo,
                                          customerIdArg,
                                          new CustomerServiceClientBinder());
        return getCustomerInfoCommand;
    }

    private static async Task getCustomerInfo(int customerId, CustomerServiceClient client)
    {
        var request = new GetCustomerInfoRequest { CustomerId = customerId };
        var response = await client.GetCustomerInfoAsync(request);

        Console.WriteLine(response);
    }
}
